modules/javafx.base/src/main/java/javafx/beans/value/ObservableValue.java

```diff
@@ -137,4 +139,102 @@
      * @return The current value
      */
     T getValue();
+
+    /**
+     * Creates an {@code ObservableValue} that holds the result of applying a
+     * mapping on this {@code ObservableValue}'s value. The result is updated
+     * when this {@code ObservableValue}'s value changes. If this value is
+     * {@code null}, no mapping is applied and the resulting value is also
+     * {@code null}.
+     * <p>
+     * For example, mapping a string to an upper case string:
+     * <pre>
+     * var text = new SimpleStringProperty("abcd");
+     * ObservableValue&lt;String&gt; upperCase = text.map(String::toUpperCase);
+     *
+     * upperCase.getValue();  // Returns "ABCD"
+     * text.set("xyz");
+     * upperCase.getValue();  // Returns "XYZ"
+     * text.set(null);
+     * upperCase.getValue();  // Returns null</pre>
+     *
+     * @param <U> the type of values held by the resulting {@code ObservableValue}
+     * @param mapper a {@code Function} that converts a given value to a new value, cannot be {@code null}
+     * @return an {@code ObservableValue} holding the result of mapping this {@code ObservableValue}'s
+     *     value, or {@code null} when it is {@code null}; never returns {@code null}
+     */
+    default <U> ObservableValue<U> map(Function<? super T, ? extends U> mapper) {
+        return new MappedBinding<>(this, mapper);
+    }
+
+    /**
+     * Creates an {@code ObservableValue} that holds this {@code ObservableValue}'s
+     * value, or the given value if it is {@code null}. The result is updated when
+     * this {@code ObservableValue}'s value changes. This method, when combined with
+     * {@link #map(Function)}, allows handling of all values including {@code null} values.
+     * <p>
+     * For example, mapping a string to an upper case string, but leaving it blank
+     * if the input is {@code null}:
+     * <pre>
+     * var text = new SimpleStringProperty("abcd");
+     * ObservableValue&lt;String&gt; upperCase = text.map(String::toUpperCase).orElse("");
+     *
+     * upperCase.getValue();  // Returns "ABCD"
+     * text.set(null);
+     * upperCase.getValue();  // Returns ""</pre>
+     *
+     * @param constant an alternative value to use when this {@code ObservableValue}
+     *     holds {@code null}; can be {@code null}
+     * @return an {@code ObservableValue} holding this {@code ObservableValue}'s value,
+     *     or the given value if it is {@code null}; never returns {@code null}
+     */
+    default ObservableValue<T> orElse(T constant) {
+        return new OrElseBinding<>(this, constant);
+    }
+
+    /**
+     * Creates an {@code ObservableValue} that holds the value of an {@code ObservableValue}
+     * resulting from applying a mapping on this {@code ObservableValue}'s value. The result
```

Comment:
I've changed this to use your wording as I think it does read much better.

Perhaps also possible:

      Creates a new {@code ObservableValue} that holds the value of a nested {@code ObservableValue} supplied
      by the given mapping function.

?

Reply:
Both seem fine, I don't have any preference over one or the other.

Reply:
I struggled with finding a good description here [previously](https://github.com/openjdk/jfx/pull/675#discussion_r777801130). I think that mstr2 gave a good approach. What we can do if we want to have "the best of both worlds" is to write something in this form:
```
<Simple slightly-inaccurate summary>. More precisely, <Correct and more convoluted description>
```

I would offer something like this based on your suggestions:

```
Creates a new {@code ObservableValue} that holds the value of a nested {@code ObservableValue} supplied by the
given mapping function. The result is updated when either this or the nested {@code ObservableValue} changes.
If either this or the nested value is {@code null}, the resulting value is {@code null} (no mapping is applied if
this value is {@code null}).
More precisely, the created {@code ObservableValue} holds the value of an {@code ObservableValue} resulting
from applying a mapping on this {@code ObservableValue}'s value.
```
I'm honestly not sure the "More precisely" part is even needed at his point. Up to you.

The `@return` description can be changed accordingly with the simplified explanation if you think it's clearer.

You can also specify a `@throws` NPE if the mapping function parameter is `null` instead of writing "cannot be null", like mstr2 suggested in another place if you like this pattern.

By the way, if we change "Creates an..." to "Creates a new..." we should change it in the other methods. I don't think there's a difference.

Reply:
Resolving this one as we have this discussion elsewhere as well.